### PR TITLE
Release GIL in convolution functions.

### PR DESCRIPTION
**Pull request by rabernat**
In response to #3949. All convolution tests pass.

### CODE REVIEW

**Comment by mdboom** [comment]
:+1:

File: CHANGES.rst [hunk 1]
```diff
@@ -16,6 +16,8 @@ New Features
 
 - ``astropy.convolution``
 
+  - Modified cython functions to release the GIL. [#3949]
```
Comment (embray) [inline CHANGES.rst]: This is a pretty nice change and deserves a better changelog entry.  First "cython" -> "Cython", but also might want to note that this enables convolution to be parallelized effectively.

Comment (embray) [inline CHANGES.rst]: Actually, could you move this to the v1.0.4 changelog section instead?  There's no reason this can't be in that release.

**Comment by embray** [comment]
Excellent, thanks!

**Comment by keflavich** [comment]
Regarding #3949, did the FFT version of convolution also perform poorly?

**Comment by rabernat** [comment]
I did not test that. If it is built on numpy.fft, it should release the GIL
and vectorize correctly.

**Pull request merged**